index.js

```diff
@@ -34,33 +37,36 @@ module.exports = function (options, cb) {
 		0x00, 0x02, /* Type:  NS      */
 		0x00, 0x01  /* Class: IN      */
 	]);
+}
 
-	var udpSocket = dgram.createSocket('udp4');
-
-	udpSocket.on('message', function (msg, rinfo) {
+function subscribeUdpSocketOnMessageReceived(udpSocket, server, options, resolve) {
+	udpSocket.on('message', (msg, rinfo) => {
+		udpSocket.close();
+		udpSocket = null;
 		if (msg && msg.length >= 2 && rinfo.address === server) {
 			// We got an answer where the source matches the queried server,
 			// we're online with high confidence
-			cb(null, true);
+			resolve(true);
 		} else {
 			// We got an answer, but it appears to not come from the queried
 			// server. Try connecting to our hostnames on port 80 and if one
 			// handshake succeeds, we're definitely online
-			isReachable(options.hostnames, cb);
+			resolve(isReachable(options.hostnames));
 		}
-
-		udpSocket.close();
-		udpSocket = null;
 	});
+}
 
-	udpSocket.send(payload, 0, payload.length, 53, server, function () {
-		setTimeout(function () {
-			// We ran into the timeout, we're offline with high confidence
-			cb(null, false);
+function sendPackage(udpSocket, server, options, resolve) {
```

Comment:
"package" is not the right terminology for UDP datagrams. Why not just call it `send`?